pkg/asset/cluster/tfvars/tfvars.go

```diff
@@ -501,11 +495,15 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		}
 		// Based on the number of workers, we could have the following outcomes:
 		// 1. workers > 0, masters not schedulable, valid cluster
-		// 2. workers = 0, masters schedulable, valid compact cluster but currently unsupported
+		// 2. workers != compute replicas (possibly caused by worker manifests being manipulated after generation), masters schedulable, invalid cluster
 		// 3. workers = 0, masters not schedulable, invalid cluster
-		if len(workers) == 0 {
-			return errors.Errorf("compact clusters with 0 workers are not supported at this time")
+		if int64(len(workers)) != numWorkers {
+			return fmt.Errorf("number of worker assets: %d does not match requested workers: %d", len(workers), numWorkers)
+		}
+		if len(workers) == 0 && !mastersSchedulable {
```

Comment:
```suggestion
		if len(workers) == 0 || !mastersSchedulable {
```
Should this be an or here?

Reply:
The condition is correct. I have updated the comment to hopefully make the current checks clearer.